src/providers/baremetal.rs

```diff
@@ -41,9 +41,12 @@ impl Setup {
         username: Option<String>,
     ) -> Result<Self, Error> {
         let username: Result<String, Error> = username.map(Ok).unwrap_or_else(|| {
-            Ok(String::from_utf8(
-                std::process::Command::new("whoami").output()?.stdout,
-            )?)
+            let user = String::from_utf8(std::process::Command::new("whoami").output()?.stdout)?;
+            let user = user
+                .split_whitespace()
```

Comment:
This will probably cause issues for people with spaces in their usernames. Perhaps `.lines()` is better? Or perhaps just `.trim()`?

Reply:
aha, I knew I was forgetting the right name for `trim()`...